packages/tenderly-core/src/internal/core/services/TenderlyService.ts

```diff
@@ -29,13 +30,21 @@ export class TenderlyService {
   }
 
   public async getNetworks(): Promise<TenderlyNetwork[]> {
+    logger.debug("Obtaining public networks...");
+
     let tenderlyApi = TenderlyApiService.configureAnonymousInstance();
     if (TenderlyApiService.isAuthenticated()) {
       tenderlyApi = TenderlyApiService.configureInstance();
     }
 
     try {
       const res = await tenderlyApi.get("/api/v1/public-networks");
+      if (res.data === undefined || res.data === null) {
+        logger.error("There was an error while obtaining public networks from Tenderly. Obtained response is invalid.");
+        return [];
+      }
+      logger.silly("Obtained public networks:", res.data);
```

Comment:
Be careful when logging full response data, not to expose any sensitive data from the `res.data`. I would prefer at least some "filtering" before logging the actual response data value. Makes sense?
I guess this API is not returning any sensitive information, but it does not matter, if for some reason it gets changed in the future to return sensitive data (since this is authenticated endpoint I guess, it is possible) you are going to leak this data with this log. Just make sure you address this...you get my point.